claims.go

```diff
@@ -8,8 +8,15 @@ import (
 
 // For a type to be a Claims object, it must just have a Valid method that determines
 // if the token is invalid for any supported reason
+// opts will often be nil
 type Claims interface {
-	Valid() error
+	Valid(opts *ValidationOptions) error
```

Comment:
This interface would stay the same.

Reply:
This interface should change IMO leeway is not part of claims it is part of validation of claims.  In-fact you can even add options like how pyjwt provides like `verify_iat : false` etc. to skip iat verify.